### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by k-takata**

### CODE REVIEW

File: docs/reporting.rst [hunk 1]
```diff
@@ -97,7 +97,7 @@ U-ctags developers know the programming language that you are talking.
     language object you asked tagging especially about kind.  A person
     extending a parser have to decide a kind of newly tagging language
     object: reusing an existing kind or introducing a new kind.
-    U-ctags developers expect a report know the concent kind, field,
+    U-ctags developers expect a report know the concept kind, field,
```
Comment (k-takata) [inline docs/reporting.rst]: Is this correct?

File: parsers/cxx/cxx_parser_function.c [hunk 2]
```diff
@@ -1816,7 +1816,7 @@ bool cxxParserTokenChainLooksLikeFunctionParameterList(
 	CXX_DEBUG_ENTER();
 	CXX_DEBUG_ASSERT(
 			tc->iCount >= 2,
-			"At least linitial and final parenthesis should be there"
+			"At least initial and final parenthesis should be there"
```
Comment (k-takata) [inline parsers/cxx/cxx_parser_function.c]: Is this correct?

File: parsers/rpmspec.c [hunk 3]
```diff
@@ -136,7 +136,7 @@ static bool found_macro_cb_full (const char *line,
 		if (data->rindex == ROLE_DEFINITION_INDEX && ctx->rejecting)
 		{
 			/* The line is continued. Let's record the cork index
-			   for attachend endLine field later. */
+			   for attached endLine field later. */
```
Comment (k-takata) [inline parsers/rpmspec.c]: Is this correct?

**Comment by k-takata** [comment]
I think I also found additional mistakes:
https://github.com/universal-ctags/ctags/blob/1258adfc11154a338d332851474e75e05ed5b39f/main/routines.c#L792
Is `luser` a mistake of `user`?
  

https://github.com/universal-ctags/ctags/blob/1258adfc11154a338d332851474e75e05ed5b39f/parsers/cxx/cxx_debug.c#L171
What does `structg` mean? Is this correct?

**Comment by masatake** [comment]
Sorry to be late. I will review this one next.

File: docs/reporting.rst (hunk 1 above)
Comment (masatake) [reply docs/reporting.rst]: Yes, your fix is correct. Thank you.

File: parsers/cxx/cxx_parser_function.c (hunk 2 above)
Comment (masatake) [reply parsers/cxx/cxx_parser_function.c]: Yes.

File: parsers/rpmspec.c (hunk 3 above)
Comment (masatake) [reply parsers/rpmspec.c]: It should be "attaching".

**Comment by masatake** [comment]
(I got an idea to run spell checker at CI environment.)

**Comment by masatake** [comment]
Thank you very much.

File: parsers/rpmspec.c (hunk 3 above)
Comment (k-takata) [reply parsers/rpmspec.c]: Updated.

File: main/entry.c [hunk 4]
```diff
@@ -1190,7 +1190,7 @@ static bool isTagWritable(const tagEntryInfo *const tag)
 			return false;
 
 		/* TODO: optimization
-		   A Bitmasks represeting all enabled roles can be calculated at the
+		   A Bitmasks represetting all enabled roles can be calculated at the
```
Comment (masatake) [inline main/entry.c]: What I would like to write was "representing".

**Comment by masatake** [comment]
@k-takata, I'm sorry but I found one more item I would like you to fix.

File: main/entry.c (hunk 4 above)
Comment (k-takata) [reply main/entry.c]: Updated.

**Comment by k-takata** [comment]
How about this: https://github.com/universal-ctags/ctags/pull/2135#issuecomment-509600876?

**Comment by masatake** [comment]
"structg" should be "struct".
"luser" may be "user" though there is a word "luser" in hacker's dictionary.

**Comment by k-takata** [comment]
Okay, fixed.

**Comment by masatake** [comment]
Thank you!

**Pull request merged**